src/profile/sagas.js

```diff
@@ -1,46 +1,36 @@
-import { push } from 'connected-react-router';
-import { all, call, delay, put, select, takeEvery } from 'redux-saga/effects';
-import { logAPIErrorResponse } from '@edx/frontend-logging';
 import { FETCH_USER_ACCOUNT_FAILURE } from '@edx/frontend-auth';
-
-// Actions
+import { App } from '@edx/frontend-base';
+import { logAPIErrorResponse } from '@edx/frontend-logging';
+import pick from 'lodash.pick';
+import { all, call, delay, put, select, takeEvery } from 'redux-saga/effects';
 import {
-  FETCH_PROFILE,
+  closeForm,
+  deleteProfilePhotoBegin,
+  deleteProfilePhotoReset,
+  deleteProfilePhotoSuccess,
+  DELETE_PROFILE_PHOTO,
   fetchProfileBegin,
-  fetchProfileSuccess,
   fetchProfileReset,
-  SAVE_PROFILE,
+  fetchProfileSuccess,
+  FETCH_PROFILE,
+  resetDrafts,
   saveProfileBegin,
-  saveProfileSuccess,
   saveProfileFailure,
-  saveProfileReset,
-  closeForm,
-  SAVE_PROFILE_PHOTO,
   saveProfilePhotoBegin,
-  saveProfilePhotoSuccess,
   saveProfilePhotoFailure,
   saveProfilePhotoReset,
-  DELETE_PROFILE_PHOTO,
-  deleteProfilePhotoBegin,
-  deleteProfilePhotoSuccess,
-  deleteProfilePhotoReset,
-  resetDrafts,
+  saveProfilePhotoSuccess,
+  saveProfileReset,
+  saveProfileSuccess,
+  SAVE_PROFILE,
+  SAVE_PROFILE_PHOTO,
 } from './actions';
-
-// Selectors
-import { handleSaveProfileSelector, handleFetchProfileSelector } from './selectors';
-
-// Services
+import { handleSaveProfileSelector, userAccountSelector } from './selectors';
 import * as ProfileApiService from './services';
 
-// Utils
-import { utils } from '../common';
-
-const { keepKeys } = utils;
-
 export function* handleFetchProfile(action) {
-  const { username } = action.payload;
-  const { authenticationUsername, userAccount } = yield select(handleFetchProfileSelector);
+  const { username, isAuthenticatedUserProfile } = action.payload;
+  const userAccount = yield select(userAccountSelector);
```

Comment:
Could we know `isAuthenticatedUserProfile` from this `select` instead of passed in through the action?

Reply:
We can't.  It's not calculated in the selectors anymore because some of its constituent data isn't in redux anymore (the authenticationUsername)